File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinOrderBookResponse.java
package com.chess.cryptobot.model.response.livecoin;

import com.chess.cryptobot.model.Price;
import com.chess.cryptobot.model.response.OrderBookResponse;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class LivecoinOrderBookResponse extends LivecoinResponse implements OrderBookResponse {
    @SerializedName("asks")
    @Expose
    public List<List<String>> asks;
    @SerializedName("bids")
    @Expose
    public List<List<String>> bids;

    @Override
    public List<Price> bids() {
        return parseValues(bids);
    }


    @Override
    public List<Price> asks() {
        return parseValues(asks);
    }

    private List<Price> parseValues(List<List<String>> values) {
        List<Price> prices = new ArrayList<>();
        values.forEach(value -> prices.add(new Price(Double.valueOf(value.get(0)), Double.valueOf(value.get(1)))));
        return prices;
    }
}

File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinOrderBookResponse.kt
package com.chess.cryptobot.model.response.livecoin

import com.chess.cryptobot.model.Price
import com.chess.cryptobot.model.response.OrderBookResponse
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class LivecoinOrderBookResponse : LivecoinResponse(), OrderBookResponse {
    @SerializedName("asks")
    @Expose
    private val asks: List<List<String>>? = null

    @SerializedName("bids")
    @Expose
    private val bids: List<List<String>>? = null
    override fun bids(): List<Price?>? {
        return parseValues(bids)
    }

    override fun asks(): List<Price?>? {
        return parseValues(asks)
    }

    private fun parseValues(values: List<List<String>>?): List<Price?> {
        val prices: MutableList<Price?> = ArrayList()
        values!!.forEach{ value: List<String> -> prices.add(Price(java.lang.Double.valueOf(value[0]), java.lang.Double.valueOf(value[1]))) }
        return prices
    }
}